common/src/View/ChangeEntityAttributesCommand.cpp

```diff
@@ -106,10 +154,22 @@ namespace TrenchBroom {
         }
 
         bool ChangeEntityAttributesCommand::doCollateWith(UndoableCommand* command) {
-            ChangeEntityAttributesCommand* other = static_cast<ChangeEntityAttributesCommand*>(command);
+            auto* other = dynamic_cast<ChangeEntityAttributesCommand*>(command);            
```

Comment:
It's checked with a different sort of check in UndoableCommand::collateWith. `if (command->type() != m_type)` which could easily be wrong in the face of bugs.

I'm not convinced we have any class hierarchy static_casts's that are justified according to 
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c146-use-dynamic_cast-where-class-hierarchy-navigation-is-unavoidable

Reply:
I find this to be overly defensive and misleading, because we know what the type must be, and state so by using static_cast. I also don't agree with every CPP core guideline ;-)

Reply:
Uhm, should have added: This is really a nitpick, of course it's fine to keep this code. I suppose we should try to find agreement on this topic so that we can make the use of casts more consistent in the future.

Reply:
Yeah, it's my bad for trying to force a coding convention change into an unrelated PR, I'll just undo it here.

Reply:
How about creating a separate issue for this. Then we can discuss it there and if we decide to to change this, we can do it consistently in one PR. Should be easy to find all downcasts with a regex.